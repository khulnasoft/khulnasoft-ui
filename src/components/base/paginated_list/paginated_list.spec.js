import { shallowMount } from '@vue/test-utils';
import GlPagination from '../pagination/pagination.vue';
import GlSearchBoxByType from '../search_box_by_type/search_box_by_type.vue';
import PaginatedList from './paginated_list.vue';

describe('Paginated List', () => {
  let wrapper;
  let template;

  beforeAll(() => {
    template = `
      <div class="slot" slot-scope="{ listItem }">
        <span class="item">Item Name: {{listItem.id}}</span>
      </div>
    `;
  });

  const createComponent = (props = {}) => {
    wrapper = shallowMount(PaginatedList, {
      scopedSlots: {
        default: template,
      },
      propsData: {
        list: [],
        ...props,
      },
    });
  };

  describe('Searchless states', () => {
    it('renders the list in an empty state, when no items are provided', () => {
      createComponent();

      expect(wrapper.findAll('.item').length).toEqual(0);
      expect(wrapper.findAll('.empty-message').length).toEqual(1);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders the list in an empty state with a custom empty message, when one is provided', () => {
      const msg = 'List is passed in empty';

      createComponent({ emptyMessage: msg });

      expect(wrapper.findAll('.item').length).toEqual(0);
      expect(wrapper.find('.empty-message').text()).toEqual(msg);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders the list with nested items, when an array of items is provided', () => {
      createComponent({
        list: [{ id: 'foo' }, { id: 'bar' }],
      });

      expect(wrapper.findAll('.item').length).toEqual(2);
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('Search states', () => {
    const list = [{ id: 'foo' }, { id: 'bar' }, { id: 'baz' }];

    it('renders the list filtered by search results', async () => {
      createComponent({ list });

      const search = wrapper.findComponent(GlSearchBoxByType);
      search.vm.$emit('input', 'ba');

      await wrapper.vm.$nextTick();
      expect(wrapper.findAll('.item').length).toBe(2);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders the default empty search message when there are no search results', async () => {
      createComponent({ list });

      const search = wrapper.findComponent(GlSearchBoxByType);
      search.vm.$emit('input', 'qux');

      await wrapper.vm.$nextTick();
      expect(wrapper.findAll('.item').length).toEqual(0);
      expect(wrapper.findAll('.empty-search').length).toEqual(1);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders the custom empty search message when there are no search results', async () => {
      const msg = 'Custom empty message';

      createComponent({ list, emptySearchMessage: msg });

      const search = wrapper.findComponent(GlSearchBoxByType);
      search.vm.$emit('input', 'qux');

      await wrapper.vm.$nextTick();
      expect(wrapper.findAll('.item').length).toEqual(0);
      expect(wrapper.find('.empty-search').text()).toEqual(msg);
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('Pagination', () => {
    const list = [
      { id: 'foo' },
      { id: 'bar' },
      { id: 'baz' },
      { id: 'qux' },
      { id: 'quux' },
      { id: 'corge' },
      { id: 'grault' },
      { id: 'garply' },
      { id: 'waldo' },
      { id: 'fred' },
      { id: 'xyzzy' },
      { id: 'plugh' },
      { id: 'thud' },
    ];

    it('renders 10 items for a default page size of 10, 13 total', () => {
      createComponent({ list });

      expect(wrapper.findAll('.item').length).toEqual(10);
      expect(wrapper.findComponent(GlPagination).props('totalItems')).toEqual(13);
      expect(wrapper.findComponent(GlPagination).props('perPage')).toEqual(10);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders 13 items for a default page size of 20', () => {
      createComponent({ list, perPage: 20 });

      expect(wrapper.findAll('.item').length).toEqual(13);
      expect(wrapper.findComponent(GlPagination).props('totalItems')).toEqual(13);
      expect(wrapper.findComponent(GlPagination).props('perPage')).toEqual(20);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders 5 items on page 1 for a page size of 5, 13 total itmes and renders pagination', () => {
      createComponent({ list, perPage: 5 });

      expect(wrapper.findAll('.item').length).toEqual(5);
      expect(wrapper.findComponent(GlPagination).props('totalItems')).toEqual(13);
      expect(wrapper.findComponent(GlPagination).props('perPage')).toEqual(5);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders 3 items on page 2 with default page size of 10 and 13 total items', () => {
      createComponent({ list, page: 2 });

      expect(wrapper.findAll('.item').length).toEqual(3);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders 1 item on page 7 with page size of 2 and 13 total items', () => {
      createComponent({ list, page: 7, perPage: 2 });

      expect(wrapper.findAll('.item').length).toEqual(1);
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('props', () => {
    describe('filterable', () => {
      it('has search enabled by default', () => {
        createComponent();

        expect(wrapper.findComponent(GlSearchBoxByType).exists()).toBe(true);
      });

      it('has search disabled when filterable prop set to false', () => {
        createComponent({ filterable: false });

        expect(wrapper.findAll('[aria-label="Search"]').length).toEqual(0);
      });
    });

    describe('filter', () => {
      const list = [{ id: 'foo' }, { id: 'bar' }];

      it('filters on default "id" key', async () => {
        createComponent({ list });

        wrapper.setData({ queryStr: 'ba' });

        await wrapper.vm.$nextTick();
        expect(wrapper.findAll('.item').length).toEqual(1);
        expect(wrapper.element).toMatchSnapshot();
      });

      it('filters on provided "myKey" key', async () => {
        createComponent({ list: [{ myKey: 'foo' }, { myKey: 'bar' }], filter: 'myKey' });

        wrapper.setData({ queryStr: 'ba' });

        await wrapper.vm.$nextTick();
        expect(wrapper.findAll('.item').length).toEqual(1);
        expect(wrapper.element).toMatchSnapshot();
      });

      it('filters with provided filter function', async () => {
        const filter = (item, queryStr) => item.id.includes(queryStr);

        createComponent({ list, filter });

        wrapper.setData({ queryStr: 'ba' });

        await wrapper.vm.$nextTick();
        const elem = wrapper.findAll('.item');
        expect(elem.length).toEqual(1);
        expect(elem.at(0).text()).toBe('Item Name: bar');
        expect(wrapper.element).toMatchSnapshot();
      });
    });
  });
});
