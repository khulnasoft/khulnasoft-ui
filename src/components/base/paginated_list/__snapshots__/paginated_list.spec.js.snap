// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginated List Pagination renders 1 item on page 7 with page size of 2 and 13 total items 1`] = `
<div>
  <div
    class="gl-flex-row-reverse gl-justify-between sm:gl-flex"
  >
    <gl-search-box-by-type-stub
      clearbuttontitle="Clear"
      value=""
    />
  </div>
  <ul
    class="gl-m-0 gl-grid gl-list-none gl-grid-cols-1 gl-divide-x-0 gl-divide-y gl-divide-solid gl-divide-default gl-p-0"
  >
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: thud
        </span>
      </div>
    </li>
  </ul>
  <gl-pagination-stub
    align="center"
    ellipsistext="…"
    labelfirstpage="Go to first page"
    labellastpage="Go to last page"
    labelnav="Pagination"
    labelnextpage="Go to next page"
    labelpage="Go to page %{page}"
    labelprevpage="Go to previous page"
    limits="[object Object]"
    nexttext="Next"
    perpage="2"
    prevtext="Previous"
    totalitems="13"
    value="7"
  />
</div>
`;

exports[`Paginated List Pagination renders 3 items on page 2 with default page size of 10 and 13 total items 1`] = `
<div>
  <div
    class="gl-flex-row-reverse gl-justify-between sm:gl-flex"
  >
    <gl-search-box-by-type-stub
      clearbuttontitle="Clear"
      value=""
    />
  </div>
  <ul
    class="gl-m-0 gl-grid gl-list-none gl-grid-cols-1 gl-divide-x-0 gl-divide-y gl-divide-solid gl-divide-default gl-p-0"
  >
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: xyzzy
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: plugh
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: thud
        </span>
      </div>
    </li>
  </ul>
  <gl-pagination-stub
    align="center"
    ellipsistext="…"
    labelfirstpage="Go to first page"
    labellastpage="Go to last page"
    labelnav="Pagination"
    labelnextpage="Go to next page"
    labelpage="Go to page %{page}"
    labelprevpage="Go to previous page"
    limits="[object Object]"
    nexttext="Next"
    perpage="10"
    prevtext="Previous"
    totalitems="13"
    value="2"
  />
</div>
`;

exports[`Paginated List Pagination renders 5 items on page 1 for a page size of 5, 13 total itmes and renders pagination 1`] = `
<div>
  <div
    class="gl-flex-row-reverse gl-justify-between sm:gl-flex"
  >
    <gl-search-box-by-type-stub
      clearbuttontitle="Clear"
      value=""
    />
  </div>
  <ul
    class="gl-m-0 gl-grid gl-list-none gl-grid-cols-1 gl-divide-x-0 gl-divide-y gl-divide-solid gl-divide-default gl-p-0"
  >
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: foo
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: bar
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: baz
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: qux
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: quux
        </span>
      </div>
    </li>
  </ul>
  <gl-pagination-stub
    align="center"
    ellipsistext="…"
    labelfirstpage="Go to first page"
    labellastpage="Go to last page"
    labelnav="Pagination"
    labelnextpage="Go to next page"
    labelpage="Go to page %{page}"
    labelprevpage="Go to previous page"
    limits="[object Object]"
    nexttext="Next"
    perpage="5"
    prevtext="Previous"
    totalitems="13"
    value="1"
  />
</div>
`;

exports[`Paginated List Pagination renders 10 items for a default page size of 10, 13 total 1`] = `
<div>
  <div
    class="gl-flex-row-reverse gl-justify-between sm:gl-flex"
  >
    <gl-search-box-by-type-stub
      clearbuttontitle="Clear"
      value=""
    />
  </div>
  <ul
    class="gl-m-0 gl-grid gl-list-none gl-grid-cols-1 gl-divide-x-0 gl-divide-y gl-divide-solid gl-divide-default gl-p-0"
  >
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: foo
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: bar
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: baz
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: qux
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: quux
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: corge
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: grault
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: garply
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: waldo
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: fred
        </span>
      </div>
    </li>
  </ul>
  <gl-pagination-stub
    align="center"
    ellipsistext="…"
    labelfirstpage="Go to first page"
    labellastpage="Go to last page"
    labelnav="Pagination"
    labelnextpage="Go to next page"
    labelpage="Go to page %{page}"
    labelprevpage="Go to previous page"
    limits="[object Object]"
    nexttext="Next"
    perpage="10"
    prevtext="Previous"
    totalitems="13"
    value="1"
  />
</div>
`;

exports[`Paginated List Pagination renders 13 items for a default page size of 20 1`] = `
<div>
  <div
    class="gl-flex-row-reverse gl-justify-between sm:gl-flex"
  >
    <gl-search-box-by-type-stub
      clearbuttontitle="Clear"
      value=""
    />
  </div>
  <ul
    class="gl-m-0 gl-grid gl-list-none gl-grid-cols-1 gl-divide-x-0 gl-divide-y gl-divide-solid gl-divide-default gl-p-0"
  >
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: foo
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: bar
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: baz
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: qux
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: quux
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: corge
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: grault
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: garply
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: waldo
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: fred
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: xyzzy
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: plugh
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: thud
        </span>
      </div>
    </li>
  </ul>
  <gl-pagination-stub
    align="center"
    ellipsistext="…"
    labelfirstpage="Go to first page"
    labellastpage="Go to last page"
    labelnav="Pagination"
    labelnextpage="Go to next page"
    labelpage="Go to page %{page}"
    labelprevpage="Go to previous page"
    limits="[object Object]"
    nexttext="Next"
    perpage="20"
    prevtext="Previous"
    totalitems="13"
    value="1"
  />
</div>
`;

exports[`Paginated List Search states renders the custom empty search message when there are no search results 1`] = `
<div>
  <div
    class="gl-flex-row-reverse gl-justify-between sm:gl-flex"
  >
    <gl-search-box-by-type-stub
      clearbuttontitle="Clear"
      value=""
    />
  </div>
  <ul
    class="gl-m-0 gl-grid gl-list-none gl-grid-cols-1 gl-divide-x-0 gl-divide-y gl-divide-solid gl-divide-default gl-p-0"
  />
  <div
    class="bs-callout bs-callout-warning empty-message gl-mt-5 empty-search"
  >
    Custom empty message
  </div>
</div>
`;

exports[`Paginated List Search states renders the default empty search message when there are no search results 1`] = `
<div>
  <div
    class="gl-flex-row-reverse gl-justify-between sm:gl-flex"
  >
    <gl-search-box-by-type-stub
      clearbuttontitle="Clear"
      value=""
    />
  </div>
  <ul
    class="gl-m-0 gl-grid gl-list-none gl-grid-cols-1 gl-divide-x-0 gl-divide-y gl-divide-solid gl-divide-default gl-p-0"
  />
  <div
    class="bs-callout bs-callout-warning empty-message gl-mt-5 empty-search"
  >
    Sorry, your filter produced no results.
  </div>
</div>
`;

exports[`Paginated List Search states renders the list filtered by search results 1`] = `
<div>
  <div
    class="gl-flex-row-reverse gl-justify-between sm:gl-flex"
  >
    <gl-search-box-by-type-stub
      clearbuttontitle="Clear"
      value=""
    />
  </div>
  <ul
    class="gl-m-0 gl-grid gl-list-none gl-grid-cols-1 gl-divide-x-0 gl-divide-y gl-divide-solid gl-divide-default gl-p-0"
  >
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: bar
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: baz
        </span>
      </div>
    </li>
  </ul>
  <gl-pagination-stub
    align="center"
    ellipsistext="…"
    labelfirstpage="Go to first page"
    labellastpage="Go to last page"
    labelnav="Pagination"
    labelnextpage="Go to next page"
    labelpage="Go to page %{page}"
    labelprevpage="Go to previous page"
    limits="[object Object]"
    nexttext="Next"
    perpage="10"
    prevtext="Previous"
    totalitems="2"
    value="1"
  />
</div>
`;

exports[`Paginated List Searchless states renders the list in an empty state with a custom empty message, when one is provided 1`] = `
<div>
  <div
    class="gl-flex-row-reverse gl-justify-between sm:gl-flex"
  >
    <gl-search-box-by-type-stub
      clearbuttontitle="Clear"
      value=""
    />
  </div>
  <ul
    class="gl-m-0 gl-grid gl-list-none gl-grid-cols-1 gl-divide-x-0 gl-divide-y gl-divide-solid gl-divide-default gl-p-0"
  />
  <div
    class="bs-callout bs-callout-warning empty-message gl-mt-5 empty-message"
  >
    List is passed in empty
  </div>
</div>
`;

exports[`Paginated List Searchless states renders the list in an empty state, when no items are provided 1`] = `
<div>
  <div
    class="gl-flex-row-reverse gl-justify-between sm:gl-flex"
  >
    <gl-search-box-by-type-stub
      clearbuttontitle="Clear"
      value=""
    />
  </div>
  <ul
    class="gl-m-0 gl-grid gl-list-none gl-grid-cols-1 gl-divide-x-0 gl-divide-y gl-divide-solid gl-divide-default gl-p-0"
  />
  <div
    class="bs-callout bs-callout-warning empty-message gl-mt-5 empty-message"
  >
    There are currently no items in this list.
  </div>
</div>
`;

exports[`Paginated List Searchless states renders the list with nested items, when an array of items is provided 1`] = `
<div>
  <div
    class="gl-flex-row-reverse gl-justify-between sm:gl-flex"
  >
    <gl-search-box-by-type-stub
      clearbuttontitle="Clear"
      value=""
    />
  </div>
  <ul
    class="gl-m-0 gl-grid gl-list-none gl-grid-cols-1 gl-divide-x-0 gl-divide-y gl-divide-solid gl-divide-default gl-p-0"
  >
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: foo
        </span>
      </div>
    </li>
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: bar
        </span>
      </div>
    </li>
  </ul>
  <gl-pagination-stub
    align="center"
    ellipsistext="…"
    labelfirstpage="Go to first page"
    labellastpage="Go to last page"
    labelnav="Pagination"
    labelnextpage="Go to next page"
    labelpage="Go to page %{page}"
    labelprevpage="Go to previous page"
    limits="[object Object]"
    nexttext="Next"
    perpage="10"
    prevtext="Previous"
    totalitems="2"
    value="1"
  />
</div>
`;

exports[`Paginated List props filter filters on default "id" key 1`] = `
<div>
  <div
    class="gl-flex-row-reverse gl-justify-between sm:gl-flex"
  >
    <gl-search-box-by-type-stub
      clearbuttontitle="Clear"
      value=""
    />
  </div>
  <ul
    class="gl-m-0 gl-grid gl-list-none gl-grid-cols-1 gl-divide-x-0 gl-divide-y gl-divide-solid gl-divide-default gl-p-0"
  >
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: bar
        </span>
      </div>
    </li>
  </ul>
  <gl-pagination-stub
    align="center"
    ellipsistext="…"
    labelfirstpage="Go to first page"
    labellastpage="Go to last page"
    labelnav="Pagination"
    labelnextpage="Go to next page"
    labelpage="Go to page %{page}"
    labelprevpage="Go to previous page"
    limits="[object Object]"
    nexttext="Next"
    perpage="10"
    prevtext="Previous"
    totalitems="1"
    value="1"
  />
</div>
`;

exports[`Paginated List props filter filters on provided "myKey" key 1`] = `
<div>
  <div
    class="gl-flex-row-reverse gl-justify-between sm:gl-flex"
  >
    <gl-search-box-by-type-stub
      clearbuttontitle="Clear"
      value=""
    />
  </div>
  <ul
    class="gl-m-0 gl-grid gl-list-none gl-grid-cols-1 gl-divide-x-0 gl-divide-y gl-divide-solid gl-divide-default gl-p-0"
  >
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: 
        </span>
      </div>
    </li>
  </ul>
  <gl-pagination-stub
    align="center"
    ellipsistext="…"
    labelfirstpage="Go to first page"
    labellastpage="Go to last page"
    labelnav="Pagination"
    labelnextpage="Go to next page"
    labelpage="Go to page %{page}"
    labelprevpage="Go to previous page"
    limits="[object Object]"
    nexttext="Next"
    perpage="10"
    prevtext="Previous"
    totalitems="1"
    value="1"
  />
</div>
`;

exports[`Paginated List props filter filters with provided filter function 1`] = `
<div>
  <div
    class="gl-flex-row-reverse gl-justify-between sm:gl-flex"
  >
    <gl-search-box-by-type-stub
      clearbuttontitle="Clear"
      value=""
    />
  </div>
  <ul
    class="gl-m-0 gl-grid gl-list-none gl-grid-cols-1 gl-divide-x-0 gl-divide-y gl-divide-solid gl-divide-default gl-p-0"
  >
    <li
      class="gl-px-5 gl-py-4"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: bar
        </span>
      </div>
    </li>
  </ul>
  <gl-pagination-stub
    align="center"
    ellipsistext="…"
    labelfirstpage="Go to first page"
    labellastpage="Go to last page"
    labelnav="Pagination"
    labelnextpage="Go to next page"
    labelpage="Go to page %{page}"
    labelprevpage="Go to previous page"
    limits="[object Object]"
    nexttext="Next"
    perpage="10"
    prevtext="Previous"
    totalitems="1"
    value="1"
  />
</div>
`;
